src/index.js

```diff
@@ -62,8 +63,8 @@ export default function nodeResolve ( options = {} ) {
 		throw new Error( 'options.skip is no longer supported — you should use the main Rollup `external` option instead' );
 	}
 
-	if ( !useModule && !useMain && !useJsnext ) {
-		throw new Error( `At least one of options.module, options.main or options.jsnext must be true` );
+	if ( !useEsnext && !useModule && !useMain && !useJsnext ) {
+		throw new Error( `At least one of options.esnext, options.module, options.main or options.jsnext must be true` );
```

Comment:
I would sort them from the most to the least used: `options.main, options.module, options.jsnext or options.esnext`